app/src/main/java/com/boardgamegeek/ui/CollectionFragment.java

```diff
@@ -654,18 +658,74 @@ public void addFilter(@NonNull CollectionFilterer filter) {
 	@DebugLog
 	private void setEmptyText() {
 		if (emptyButton == null) return;
-		@StringRes int resId = R.string.empty_collection;
-		if (SyncPrefs.getLastCompleteCollectionTimestamp(getContext()) == 0L) {
-			resId = R.string.empty_collection_sync_never;
-			emptyButton.setVisibility(View.GONE);
-		} else if (hasFiltersApplied()) {
-			resId = R.string.empty_collection_filter_on;
-			emptyButton.setVisibility(View.VISIBLE);
-		} else if (!PreferencesUtils.isCollectionSetToSync(getContext())) {
-			resId = R.string.empty_collection_sync_off;
-			emptyButton.setVisibility(View.VISIBLE);
+		final Set<String> syncedStatuses = PreferencesUtils.getSyncStatuses(getContext());
+		final boolean collectionSyncOn = syncedStatuses != null && syncedStatuses.size() != 0;
+		if (collectionSyncOn) {
+			final boolean noPreviousSync = SyncPrefs.getLastCompleteCollectionTimestamp(getContext()) == 0L;
+			final boolean filterApplied = hasFiltersApplied();
+			if (noPreviousSync) {
+				setEmptyStateForNoAction(R.string.empty_collection_sync_never);
+			} else if (filterApplied) {
+				if (isAtLeastOneSyncOff(syncedStatuses, getListOfVisibleStatuses())) {
+					setEmptyStateForSettingsAction(R.string.empty_collection_filter_on_sync_partial);
+				} else {
+					setEmptyStateForReSyncAction(R.string.empty_collection_filter_on);
+				}
+			} else {
+				setEmptyStateForReSyncAction(R.string.empty_collection);
```

Comment:
In this case it might make more sense to show the settings action instead of the re-sync action. Users can always trigger a re-sync by pulling to refresh. Same on line 672. @ccomeaux Thoughts?

Reply:
Well it could happen for a new user that actually has no games in their collection, right? Anyway, I'll change both to link to settings. This means that we'll never give to the user the "re-sync" option (we're linking to settings in all cases). Are you good with that? 😄

Reply:
Yes - you can always pull-to-refresh.

Reply:
@ccomeaux sorry, had a couple of rough months. PR is up-to-date and should be ready to be merged. 😉